File: ios/App/Share/Model/ProductInfo.swift
//
//  ProductInfo.swift
//  Share
//
//  Created by Tim Fischer on 16.12.20.
//

import Foundation

struct ProductInfo: Equatable {

    let id: UInt
    let productUrl: String
    let imageUrl: String
    let name: String?
    let price: Price
    
    static func == (lhs: ProductInfo, rhs: ProductInfo) -> Bool {
        return lhs.id == rhs.id
    }
    
}

File: android/app/src/main/java/io/wantic/app/share/models/ProductInfo.kt
package io.wantic.app.share.models

import kotlinx.serialization.Serializable
import java.io.Serializable as JavaSerializable

@Serializable
data class ProductInfo(
    val id: Int,
    var name: String,
    var imageUrl: String?,
    val productUrl: String,
    var price: Float = 0.00f,
    var isFavorite: Boolean = false,
    var note: String? = null
) : JavaSerializable
